src/components/ItemFilter/FilterSettingsDialog.js

```diff
@@ -0,0 +1,202 @@
+    Radio,
+    Transfer,
+} from '@dhis2/ui'
+
+import classes from './styles/FilterSettingsDialog.module.css'
+
+const RadioOptions = ({
+    filtersSelectable,
+    updateFilterDimensionRestrictability,
+}) => (
+    <form>
+        <div className={classes.radioButton}>
+            <Radio
+                dense
+                checked={!filtersSelectable}
+                label={i18n.t('Allow filtering on all dimensions')}
+                name="radio-allow-filtering-on-all"
+                onChange={payload => {
+                    return updateFilterDimensionRestrictability(
+                        payload.value === 'true'
+                    )
+                }}
+                value={'false'}
+            />
+        </div>
+        <div className={classes.radioButton}>
+            <Radio
+                dense
+                checked={filtersSelectable}
+                label={i18n.t('Only allow filtering on specified dimensions')}
+                name="radio-restrict-filtering"
+                onChange={payload => {
+                    return updateFilterDimensionRestrictability(
+                        payload.value === 'true'
+                    )
+                }}
+                value={'true'}
+            />
+        </div>
+    </form>
+)
+
+RadioOptions.propTypes = {
+    filtersSelectable: PropTypes.bool,
+    updateFilterDimensionRestrictability: PropTypes.func,
+}
+
+const FilterSettingsDialog = ({
+    dimensions,
+    restrictFilters,
+    initiallySelectedItems,
+    onClose,
+    onConfirm,
+    open,
+}) => {
+    const [selected, setSelected] = useState(initiallySelectedItems)
+    const [filtersSelectable, setFiltersSelectable] = useState(restrictFilters)
+
+    const updateFilterDimensionRestrictability = val => {
+        if (val) {
+            const otherItems = selected.filter(
+                i => i !== DIMENSION_ID_ORGUNIT && i !== DIMENSION_ID_PERIOD
+            )
+            updateSelectedFilters([
+                DIMENSION_ID_PERIOD,
+                DIMENSION_ID_ORGUNIT,
+                ...otherItems,
+            ])
+        }
+        setFiltersSelectable(val)
+    }
+
+    const updateSelectedFilters = sItems => {
+        // ensure that orgUnit, period appear before other items
+        const periodItem = [
+            ...new Set(sItems.filter(i => i === DIMENSION_ID_PERIOD)),
+        ]
+        const orgUnitItem = [
+            ...new Set(sItems.filter(i => i === DIMENSION_ID_ORGUNIT)),
+        ]
+        const otherItems = sItems.filter(
+            i => i !== DIMENSION_ID_ORGUNIT && i !== DIMENSION_ID_PERIOD
+        )
+        return setSelected([...periodItem, ...orgUnitItem, ...otherItems])
+    }
+
+    const closeDialog = () => {
+        setSelected(initiallySelectedItems)
+        setFiltersSelectable(restrictFilters)
+        onClose()
+    }
+
+    return (
+        <>
+            {open && (
+                <Modal
+                    dataTest="dhis2-uicore-modal"
```

Comment:
dataTest prop can be removed since it isn't in use by the cypress tests.

Reply:
fixed